### PR TITLE
jeff/add-debugging-guide

### PR DESCRIPTION
**Pull request by jeffpatzer**
- Added to troubleshooting guide
- Cleaned up some console errors (MUI complaints)
- Bumped packages for security

image
Active menu item is just bold, no need for the bullet point

### CODE REVIEW

File: guides/troubleshooting.md [hunk 1]
```diff
@@ -55,12 +61,75 @@ You can also use the server logs to debug routing issues going to custom backend
...
+
+A Permalink request will skip the edge (cache) and go straight to the serverless tier. This will likely degrade performance, but does allow for verification of a function.
+
+The edge link will route through the edge.
+
+You can find both links on the detail page of a deployment.
+
+## Cache reasons
+
+We provide a header, `x-0-caching-status` to best understand why something is being cached. There is a [detailed guide](guides/caching#section_why_is_my_response_not_being_cached_) available on decipering those reasons.
```
Comment (rishi-raj-jain) [inline guides/troubleshooting.md:131]: decipering -> deciphering?

Comment (jeffpatzer) [reply guides/troubleshooting.md:131]: Nice catch

File: guides/nuxt.md [hunk 2]
```diff
@@ -315,6 +315,18 @@ export default new Router()
...
 
+## includeFiles
+
+Nuxt requires that certain resources are included in a build and deploy to have access to them. As such, at times this will require additional configuration. To include additional resources for server side rendering, API calls, etc., use the `includeFiles` option in your `layer0.config.js` file. [Read more](/guides/layer0_config#section_includefiles)
+
+In this example, we would have an `api` folder that we want to include all items from.
+
+```js
+includeFiles: {
+  'api/**/*': true,
```
Comment (rishi-raj-jain) [inline guides/nuxt.md:326]: does just 
includeFiles : {
    api: true
} 
work fine?

File: guides/nuxt.md [hunk 3]
```diff
@@ -315,6 +315,18 @@ export default new Router()
 
 If you set the `fallback` property in the [generate](https://nuxtjs.org/docs/2.x/configuration-glossary/configuration-generate/#fallback) config to `true`, Nuxt.js will generate a 404.html page that will be served whenever the URL does not match a static page. {{ PRODUCT_NAME }} will send a 404 http status for these URLs. Note that if you set the fallback property to a string, Nuxt will generate a fallback page with that name, and {{ PRODUCT_NAME }} will serve it with a 200 http status when the URL does not match a statically generated page.
 
+## includeFiles
+
+Nuxt requires that certain resources are included in a build and deploy to have access to them. As such, at times this will require additional configuration. To include additional resources for server side rendering, API calls, etc., use the `includeFiles` option in your `layer0.config.js` file. [Read more](/guides/layer0_config#section_includefiles)
```
Comment (rishi-raj-jain) [inline guides/nuxt.md:320]: maybe worth mentioning that if one wants to include some packages that aren't covered in includeNodeModules: true, that can be done via includeFiles like the following:
includeFiles: {
     'node_modules/some_pakage/**/*': true
}

File: guides/nuxt.md (hunk 2 above)
Comment (jeffpatzer) [reply guides/nuxt.md:326]: It looks like it does, but that does not really line up with the [docs](https://docs.layer0.co/guides/v3.16.5/layer0_config#section_includefiles) that we already have. I'd rather stay consistent.

File: guides/nuxt.md (hunk 3 above)
Comment (jeffpatzer) [reply guides/nuxt.md:320]: Good tip. Added.

**Comment by jeffpatzer** [comment]
@tristanlee85 can you review this? i want to get the PR merged

File: components/Toc.js [hunk 4]
```diff
@@ -161,10 +160,6 @@ function Item({ heading, activeHeading }) {
 
   return (
     <div className={classes.item} style={{ marginLeft: theme.spacing((tokens.length - 2) * 2) }}>
-      <div
-        className={classes.dot}
-        style={{ visibility: id === activeHeading ? 'visible' : 'hidden' }}
-      />
```
Comment (tristanlee85) [inline components/Toc.js:167]: This styling has been here for a while. Was it requested to be removed?

File: components/nav/Link.js [hunk 5]
```diff
@@ -76,10 +75,5 @@ export default function Link({ className, icon, text, as, href, ...props }) {
 
   const active = asPath.split(/\?/)[0] === as
 
-  return (
-    <div className={clsx({ [classes.root]: true, [classes.active]: active })}>
-      <div className={classes.dot} style={{ visibility: active ? 'visible' : 'hidden' }}></div>
-      {link}
-    </div>
-  )
```
Comment (tristanlee85) [inline components/nav/Link.js:84]: Same as above ^^

File: guides/troubleshooting.md [hunk 6]
```diff
@@ -27,15 +31,17 @@ The above assumes that the workspace folder is your app's root directory. If tha
...
-## Cloud
+## Logs
+
+{{ PRODUCT_NAME }} provides two types of logs to you debug issues with your application.
 
-{{ PRODUCT_NAME }} provides two types of logs to you debug issues with your application:
+### [Server logs](/guides/logs#section_server_logs)
 
-- [Server logs](/guides/logs#section_server_logs)
+By viewing the server logs in the {{ PRODUCT_NAME }} Developer Console, you can see all of the messages logged by your application using console.log, console.warn, etc...
```
Comment (tristanlee85) [inline guides/troubleshooting.md:40]:
```suggestion
By viewing the server logs in the {{ PRODUCT_NAME }} Developer Console, you can see all of the messages logged by your application using `console.log`, `console.warn`, etc...
```

File: guides/troubleshooting.md [hunk 7]
```diff
@@ -55,12 +61,75 @@ You can also use the server logs to debug routing issues going to custom backend
...
+
+**View headers only**
+
+```bash
+curl -o/dev/null -vv https://www.yoursite.com
+```
+
+**Bypass DNS resolution**
+
+Connect directly to the address listed after. This is good for sending a request straight to origin and bypassing L0, or testing a connection to Layer0 before DNS cutover. Setting up a localhost DNS configuration is usually better for this if possible.
```
Comment (tristanlee85) [inline guides/troubleshooting.md:88]:
```suggestion
Connect directly to the address listed after. This is good for sending a request straight to origin and bypassing Layer0, or testing a connection to Layer0 before DNS cutover. Setting up a localhost DNS configuration is usually better for this if possible.
```

File: guides/troubleshooting.md [hunk 8]
```diff
@@ -55,12 +61,75 @@ You can also use the server logs to debug routing issues going to custom backend
...
+**Send a specific user agent**
+
+```bash
+curl -o/dev/null -vv
+    -H "User-Agent: Mozilla/5.0 (iPhone; CPU iPhone OS 13_6_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148 [FBAN/FBIOS;FBDV/iPhone10,2;FBMD/iPhone;FBSN/iOS;FBSV/13.6.1;FBSS/3;FBID/phone;FBLC/en_GB;FBOP/5];FBNV/1"
+```
+
+**Skip the cache**
+
+Adding a `layer0_debug=true` to the query parameter will skip the cache and make it easy to check for dynamic data (i.e. personalized content). Append grep to search for something you might want.
```
Comment (tristanlee85) [inline guides/troubleshooting.md:113]:
```suggestion
Adding a `layer0_debug=true` to the query parameter will skip the cache and make it easy to check for dynamic data (i.e. personalized content). Append `grep` to search for something you might want.
```

> Append `grep` to search for something you might want.

Maybe something like:
> Append `grep` to search for a specific value in the response output.

(or something along those lines?)

**Review by tristanlee85 — CHANGES_REQUESTED** [review]

File: components/Toc.js (hunk 4 above)
Comment (jeffpatzer) [reply components/Toc.js:167]: Specifically, no. But I asked Mark about doing these type of updates and he said to go for it.

**Comment by jeffpatzer** [comment]
@tristanlee85 answered your comments

**Review by tristanlee85 — APPROVED** [review]

**Pull request merged**